### PR TITLE
Fix missing connection metadata

### PR DESCRIPTION
**Pull request by winder**
# Summary

When telemetry is disabled connection information is incomplete when communicating with relay nodes.

## Test Plan

Before and after the fix, started a node with telemetry disabled and see that the error is reproduced and then fixed.

Missing metadata before fix:
before_fix

Metadata after fix:
after_fix

### CODE REVIEW

**Comment by winder** [comment]
@tsachiherman updated based on your feedback. I think using a global is fine because there is a single config file. When we start the `externalized telemetry` project that may be revisited.

File: logging/telemetry.go [hunk 1]
```diff
@@ -87,6 +84,8 @@ func EnsureTelemetryConfigCreated(configDir *string, genesisID string) (Telemetr
 		var err error
 		configPath, err = config.GetConfigFilePath(loggingFilename)
 		if err != nil {
+			cfg := createTelemetryConfig()
+			initializeConfig(cfg)
 			return createTelemetryConfig(), true, err
```
Comment (tsachiherman) [inline logging/telemetry.go]: I think that you meant:
`return cfg, true, err`

Comment (winder) [reply logging/telemetry.go]: Yes, thank you.

**Comment by algobolson** [comment]
Is there any privacy implication in disabling telemetry that implies, "I don't want to be identifiable in other nodes' telemetry" ? Should a telemetry-disabled node send "" as its GUID?

**Comment by winder** [comment]
@algobolson I'm not sure. Telemetry is disabled by default, so we can't read into that fact as a user taking any specific action. I think that because the GUID is a random UUID, this is a good balance of respecting privacy and useful network diagnostic data.

**Comment by Karmastic** [comment]
> @algobolson I'm not sure. Telemetry is disabled by default, so we can't read into that fact as a user taking any specific action. I think that because the GUID is a random UUID, this is a good balance of respecting privacy and useful network diagnostic data.

Yeah this is a squishy area.  We're only using this GUID to correlate network connections from the same host (the whole point of this PR).  This doesn't leak anything that is PII.

**Pull request merged**